lighthouse-core/test/gather/gatherers/accessibility-test.js

```diff
@@ -32,3 +32,51 @@ describe('Accessibility gatherer', () => {
       err => assert.ok(err.message.includes(error)));
   });
 });
+
+describe('axe-run', () => {
+  /**
+   * @jest-environment jsdom
+   */
+  const fs = require('fs');
+  const jsdom = require('jsdom');
+  const pageFunctions = require('../../../lib/page-functions.js');
+
+  const jdomObj = new jsdom.JSDOM(`<!doctype html><meta charset="utf8"><title>hi</title>valid.`);
+  let axe;
+
+  beforeAll(() =>{
+    // Needed by axe-core
+    // https://github.com/dequelabs/axe-core/blob/581c441c/doc/examples/jsdom/test/a11y.js#L24
+    global.window = global.self = jdomObj.window;
+    global.document = jdomObj.window.document;
+    global.getNodeDetails = pageFunctions.getNodeDetails;
+
+    // axe-core must be required after the global polyfills
+    axe = require('axe-core'); // eslint-disable-line no-unused-vars
+  });
+
+  afterAll(() => {
+    global.window = undefined;
+    global.document = undefined;
+    global.getNodeDetails = undefined;
+  });
+
+  it('tests only the checks we have audits defined for', async () => {
+    const axeResults = await AccessibilityGather._runA11yChecksInTestMode();
+
+    const axeRuleIds = new Set();
+    for (const type of ['violations', 'incomplete', 'inapplicable', 'passes']) {
+      axeResults[type] && axeResults[type].forEach(result => axeRuleIds.add(result.id));
+    }
+    // The color-contrast rule is manually disabled as jsdom doesn't support the APIs needed. https://github.com/dequelabs/axe-core/blob/581c441c/doc/examples/jsdom/test/a11y.js#L40
```

Comment:
with the amount of complexity here from jsdom and distance from reality, I'm pretty close to sayin' we should just do this in puppeteer instead. WDYT?

```js
let browser;

beforeAll(async () => {
  browser = await require('puppeteer').launch();
})

afterAll(async () => {
  await browser.close()
})

it('should work', () => {
  const page = await browser.newPage()
  await page.evaluate(`${axe} ; ${getNodeDetails} ; ${daFn}`) // or intercept the string by running the gatherer itself with a `mockCommand`?
})
```